lms/djangoapps/branding/api.py

```diff
@@ -0,0 +1,214 @@
+            "math, medicine, music, philosophy, physics, science, statistics and more. {EdX} is a "
+            "non-profit online initiative created by founding partners {Harvard} and {MIT}."
+        ).format(
+            EdX="EdX", Harvard="Harvard", MIT="MIT", HarvardX="HarvardX", MITx="MITx",
+            BerkeleyX="BerkeleyX", UTx="UTx"
+        )
+    else:
+        data = ""
+    return data
+
+
+def social_links():
+    """ Returns the list of social link of footer
+    """
+    links = []
+    for social_name in settings.SOCIAL_MEDIA_FOOTER_NAMES:
+        links.append(
+            {
+                "provider": social_name,
+                "title": unicode(settings.SOCIAL_MEDIA_FOOTER_DISPLAY.get(social_name, {}).get("title", "")),
+                "url": settings.SOCIAL_MEDIA_FOOTER_URLS.get(social_name, "#"),
+                "social_icon": settings.SOCIAL_MEDIA_FOOTER_DISPLAY.get(social_name).get("icon", "")
+            }
+        )
+    return links
+
+
+def about_edx_link():
+    """ Returns the list of marketing links of footer
+    """
+
+    return [
+        {
+            "title": _("About"),
+            "url": marketing_link('ABOUT')
+        },
+        {
+            "title": _("News"),
+            "url": marketing_link('NEWS')
+        },
+        {
+            "title": _("Contact"),
+            "url": marketing_link('CONTACT')
+        },
+        {
+            "title": _("FAQ"),
+            "url": marketing_link('FAQ')
+        },
+        {
+            "title": _("edX Blog"),
+            "url": marketing_link('BLOG')
+        },
+        {
+            "title": _("Donate to edX"),
+            "url": marketing_link('DONATE')
+        },
+        {
+            "title": _("Jobs at edX"),
+            "url": marketing_link('JOBS')
+        }
+    ]
+
+
+def get_footer_logo(site_name, is_edx_domain):
+    """ Return the logo used for footer about link
+
+    Args:
+        site_name(str): The site url to get the absolute link
+        is_edx_domain(bool): Flag to check is it in edx domain or open edx
+
+    Returns:
+        Absolute url to logo
+    """
+    if is_edx_domain:
+        logo_file = 'images/edx-theme/edx-header-logo.png'
+    else:
+        logo_file = 'images/default-theme/logo.png'
+    try:
+        url = site_name + staticfiles_storage.url(logo_file)
+    except:
+        url = site_name + logo_file
+    return url
+
+
+def get_mobile_urls():
+    """ Returns the mobile app urls
+
+    """
+    mobile_urls = []
+    if settings.FEATURES.get('ENABLE_FOOTER_MOBILE_APP_LINKS'):
+        mobile_urls = [
+            {
+                "title": "Apple",
+                "url": settings.MOBILE_STORE_URLS.get('apple', '#'),
+                "logo_image_url": 'images/app/app_store_badge_135x40.svg'
+            },
+            {
+                "title": "Google",
+                "url": settings.MOBILE_STORE_URLS.get('google', '#'),
```

Comment:
Is using `#` as a default link the right behavior? If no URL is configured in settings, it seems like the object representing this link shouldn't be included in the data returned by the API.